### PR TITLE
Remove svg preamble validation

### PR DESCRIPTION
**Pull request by hashedone**
Problem occurred for technically proper svg file with no preamble - turns out that the only proper way to validate svg file is full xml parse and ensuring it is proper xml with only svg related tags (but all the cdata sections and comments are allowed). It is too complicated and to costly in gas, so for now it might be better just to check file size.

### CODE REVIEW

File: contracts/cw20-base/src/contract.rs [hunk 1]
```diff
@@ -25,33 +25,8 @@ const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
...
-
-    // Additionally attributes format could be validated as they are well defined, as well as
-    // comments presence inside of preable, but it is probably not worth it.
-}
-
 /// Validates XML logo
 fn verify_xml_logo(logo: &[u8]) -> Result<(), ContractError> {
-    verify_xml_preamble(logo)?;
-
     if logo.len() > LOGO_SIZE_CAP {
```
Comment (maurolacy) [inline contracts/cw20-base/src/contract.rs:30]:
```suggestion
    if logo.is_empty() || logo.len() > LOGO_SIZE_CAP {
```
or a min size?

**Review by maurolacy — APPROVED** [review]
Less is more. Fine with me.

File: contracts/cw20-base/src/contract.rs (hunk 1 above)
Comment (hashedone) [reply contracts/cw20-base/src/contract.rs:30]: Why? If `logo.is_empty()`, then `.len()` by definition returns `0` for slice, it is not an UB...

Comment (maurolacy) [reply contracts/cw20-base/src/contract.rs:30]: Nothing to do with UB. Just to add a check for minimum / non-empty too.

**Pull request merged**